builder/anka/step_set_generated_data.go

```diff
@@ -0,0 +1,61 @@
+package anka
+
+import (
+	"bytes"
+	"context"
+	"log"
+
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	"github.com/hashicorp/packer-plugin-sdk/packerbuilderdata"
+	"github.com/veertuinc/packer-builder-veertu-anka/client"
+)
+
+type StepSetGeneratedData struct {
+	client        client.Client
+	vmName        string
+	GeneratedData *packerbuilderdata.GeneratedData
+}
+
+func (s *StepSetGeneratedData) Run(_ context.Context, state multistep.StateBag) multistep.StepAction {
+	log.Printf("Exposing build contextual variables...")
+
+	s.client = state.Get("client").(client.Client)
+	s.vmName = state.Get("vm_name").(string)
+	osVersion := state.Get("os_version")
+	darwinVersion := client.RunParams{
+		Command: []string{"/usr/bin/uname", "-r"},
+		VMName:  s.vmName,
+		Stdout:  &bytes.Buffer{},
+	}
+
+	_, err := s.client.Run(darwinVersion)
+	if err != nil {
+		return multistep.ActionHalt
+	}
+
+	if osVersion == nil {
+		osv := client.RunParams{
+			Command: []string{"/usr/bin/sw_vers", "-productVersion"},
```

Comment:
So `sw_vers` will give back `X.Y.Z` while the Info plist only gives back `X.Y`. Do we want to drop the `.Z` on this so that they are consistent in form?